src/lib/transform-fast-rest.js

```diff
@@ -43,7 +43,12 @@ export default function fastRestTransform({ template, types: t }) {
 					func.arrowFunctionToExpression();
 				}
 
-				if (binding.constant && binding.referencePaths.length === 1) {
+				if (
+					binding.constant &&
+					binding.referencePaths.length === 1 &&
+					// arguments access requires the same function scope
+					getParentFunction(binding.referencePaths[0], t) === func
```

Comment:
nit: there's a builtin for this (though see my other comment for why you might actually have the better approach here!)

```suggestion
					binding.referencePaths[0].getFunctionParent() === func
```

Reply:
aww, that's nice!